IdentityCore/src/validation/MSIDAADAuthority.m

```diff
@@ -0,0 +1,247 @@
+    {
+        // AAD v1 doesn't support consumer authority
+        return @[];
+    }
+    
+    NSMutableArray *aliases = [NSMutableArray array];
+    
+    if (self.tenant.type == MSIDAADTenantTypeIdentifier)
+    {
+        // If it's a tenanted authority, lookup original authority and common as those are the same, but start with original authority
+        [aliases addObjectsFromArray:[self cacheAliases]];
+        
+        __auto_type aadAuthorityCommon = [MSIDAADAuthority aadAuthorityWithEnvironment:[self.url msidHostWithPortIfNecessary] rawTenant:MSIDAADTenantTypeCommonRawValue context:nil error:nil];
+        [aliases addObjectsFromArray:[aadAuthorityCommon cacheAliases]];
+    }
+    else
+    {
+        // If it's a tenantless authority, lookup by universal "common" authority, which is supported by both v1 and v2
+        [aliases addObjectsFromArray:[self cacheAliases]];
+    }
+    
+    return aliases;
+}
+
++ (BOOL)isAuthorityFormatValid:(NSURL *)url
+                       context:(id<MSIDRequestContext>)context
+                         error:(NSError **)error
+{
+    if (![super isAuthorityFormatValid:url context:context error:error]) return NO;
+    
+    __auto_type tenant = [self tenantFromAuthorityUrl:url context:context error:error];
+    
+    if ([tenant.rawTenant isEqualToString:@"adfs"])
+    {
+        if (error)
+        {
+            __auto_type message = [NSString stringWithFormat:@"Trying to initialize AAD authority with ADFS authority url."];
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, message, nil, nil, nil, context.correlationId, nil);
+        }
+        return NO;
+    }
+    
+    if ([tenant.rawTenant isEqualToString:@"tfp"])
+    {
+        if (error)
+        {
+            __auto_type message = [NSString stringWithFormat:@"Trying to initialize AAD authority with B2C authority url."];
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidDeveloperParameter, message, nil, nil, nil, context.correlationId, nil);
+        }
+        return NO;
+    }
+    
+    return tenant != nil;
+}
+
++ (instancetype)aadAuthorityWithEnvironment:(NSString *)environment
+                                   rawTenant:(NSString *)rawTenant
+                                     context:(id<MSIDRequestContext>)context
+                                       error:(NSError **)error
+{
+    __auto_type authorityUrl = [NSURL URLWithString:[NSString stringWithFormat:@"https://%@/%@", environment, rawTenant]];
+    __auto_type authority = [[MSIDAADAuthority alloc] initWithURL:authorityUrl context:context error:error];
+    
+    return authority;
+}
+
+- (nonnull NSString *)authorityType
+{
+    return MSID_TELEMETRY_VALUE_AUTHORITY_AAD;
+}
+
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone
+{
+    MSIDAADAuthority *authority = [super copyWithZone:zone];
+    authority->_tenant = [_tenant copyWithZone:zone];
```

Comment:
should we also copy openIdConfigurationEndpoint?

Reply:
done

Reply:
did it get done?